### PR TITLE
feat: add single-PR targeted reviewer and fixer discovery

### LINKED ISSUE #368: feat: add single-PR targeted reviewer and fixer discovery
## Parent

#366

## What to build

Add formal single-PR targeted discovery entrypoints for reviewer and fixer so a caller can evaluate one PR without scanning the entire repo. The new entrypoints must reuse the same runner-owned eligibility, guardrails, loop creation, and enqueue semantics as existing poll discovery.

Webhook payloads are not part of this slice. This slice creates the runner API that later webhook routing will call.

## Acceptance criteria

- [ ] Reviewer exposes a single-PR targeted discovery entrypoint that evaluates one live PR for a project/repo/PR number.
- [ ] Fixer exposes a single-PR targeted discovery entrypoint that evaluates one live PR for a project/repo/PR number.
- [ ] Targeted reviewer discovery preserves existing review-request, draft/open state, self-review, quiet-period, and active dedupe guardrails.
- [ ] Targeted fixer discovery preserves existing fix item derivation, marker checks, loop creation, fix item hashing, and active dedupe guardrails.
- [ ] Poll discovery continues to use the same shared per-PR processing logic rather than duplicating behavior.
- [ ] Tests cover eligible, skipped, and duplicate-active cases for reviewer and fixer targeted discovery.
- [ ] ok  	github.com/nexu-io/looper/cmd/looper	(cached)
ok  	github.com/nexu-io/looper/cmd/looperd	(cached)
ok  	github.com/nexu-io/looper/internal/agent	(cached)
ok  	github.com/nexu-io/looper/internal/api	(cached)
?   	github.com/nexu-io/looper/internal/app	[no test files]
ok  	github.com/nexu-io/looper/internal/bootstrap	(cached)
ok  	github.com/nexu-io/looper/internal/cliapp	(cached)
ok  	github.com/nexu-io/looper/internal/config	(cached)
ok  	github.com/nexu-io/looper/internal/coordinator	(cached)
ok  	github.com/nexu-io/looper/internal/coordinator/dispatch	(cached)
ok  	github.com/nexu-io/looper/internal/coordinator/triage	(cached)
ok  	github.com/nexu-io/looper/internal/diffanchor	(cached)
ok  	github.com/nexu-io/looper/internal/disclosure	(cached)
ok  	github.com/nexu-io/looper/internal/domain	(cached)
ok  	github.com/nexu-io/looper/internal/e2e	(cached)
ok  	github.com/nexu-io/looper/internal/e2e/githubcontract	(cached)
ok  	github.com/nexu-io/looper/internal/e2e/harness	(cached)
ok  	github.com/nexu-io/looper/internal/e2e/harness/cmd/fake-agent	(cached)
?   	github.com/nexu-io/looper/internal/e2e/harness/cmd/fake-gh	[no test files]
?   	github.com/nexu-io/looper/internal/e2e/harness/cmd/fake-osascript	[no test files]
?   	github.com/nexu-io/looper/internal/eventlog	[no test files]
ok  	github.com/nexu-io/looper/internal/fixer	(cached)
?   	github.com/nexu-io/looper/internal/infra	[no test files]
ok  	github.com/nexu-io/looper/internal/infra/git	(cached)
ok  	github.com/nexu-io/looper/internal/infra/github	(cached)
ok  	github.com/nexu-io/looper/internal/infra/notify	(cached)
ok  	github.com/nexu-io/looper/internal/infra/shell	(cached)
ok  	github.com/nexu-io/looper/internal/infra/specpr	(cached)
?   	github.com/nexu-io/looper/internal/infra/worktree	[no test files]
ok  	github.com/nexu-io/looper/internal/lifecycle	(cached)
ok  	github.com/nexu-io/looper/internal/loops	(cached)
ok  	github.com/nexu-io/looper/internal/planner	(cached)
ok  	github.com/nexu-io/looper/internal/projects	(cached)
ok  	github.com/nexu-io/looper/internal/release	(cached)
ok  	github.com/nexu-io/looper/internal/reviewer	(cached)
ok  	github.com/nexu-io/looper/internal/runs	(cached)
ok  	github.com/nexu-io/looper/internal/runtime	(cached)
?   	github.com/nexu-io/looper/internal/scheduler	[no test files]
ok  	github.com/nexu-io/looper/internal/storage	(cached)
ok  	github.com/nexu-io/looper/internal/sweeper	(cached)
ok  	github.com/nexu-io/looper/internal/version	(cached)
ok  	github.com/nexu-io/looper/internal/worker	(cached)
ok  	github.com/nexu-io/looper/internal/worktreesafety	(cached)
ok  	github.com/nexu-io/looper/pkg/api	(cached)
?   	github.com/nexu-io/looper/tools/go-build-flags	[no test files]
?   	github.com/nexu-io/looper/tools/print-version-json	[no test files]
?   	github.com/nexu-io/looper/tools/release-manifest	[no test files], , and  pass.

## Blocked by

- #367

### PR DESCRIPTION
**Pull request by mrcfps**
## Summary
- add targeted single-PR discovery entrypoints for reviewer and fixer that evaluate one live PR without scanning the repo
- refactor poll discovery to reuse shared per-PR processing helpers so targeted and poll paths keep the same loop, dedupe, and guardrail behavior
- add reviewer and fixer tests for eligible, skipped, and duplicate-active targeted discovery cases

## Validation
- go test ./...
- go build ./...

Closes #368

<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=worker · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

### CODE REVIEW

File: internal/reviewer/runner.go [hunk 1]
```diff
@@ -801,36 +747,169 @@ func (r *Runner) DiscoverPullRequests(ctx context.Context, input DiscoveryInput)
 			result.Skipped++
 			continue
 		}
-		if !policy.IncludeDrafts && detail.IsDraft {
-			result.Skipped++
-			continue
+		seen[key] = struct{}{}
```
[the remark this answers is not in the document]
Comment (mrcfps) [reply internal/reviewer/runner.go]: @chatgpt-codex-connector This has been fixed in ec83559.

Updated internal/reviewer/runner.go so follow-up discovery marks a PR as seen only after discoverExistingReviewerLoop actually appends a queue item. That preserves eligible manual follow-up loops when an earlier automatic loop is skipped.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

File: internal/reviewer/runner.go [hunk 2]
```diff
@@ -801,36 +747,169 @@ func (r *Runner) DiscoverPullRequests(ctx context.Context, input DiscoveryInput)
...
+	}
+	if existingLoop != nil {
+		if err := r.discoverExistingReviewerLoop(ctx, *project, input.Repo, policy, &currentLogin, *existingLoop, detail, &result); err != nil {
+			return DiscoveryResult{}, err
 		}
-		requireReviewRequest := requireReviewRequestForLoop(loop, policy.RequireReviewRequest, detail.HeadSHA)
-		if requireReviewRequest && !isCurrentUserRequested(detail.ReviewRequests, currentLogin) && !r.hasThreadResolutionFollowUpCandidate(ctx, project.RepoPath, input.Repo, *loop.PRNumber, detail.HeadSHA, currentLogin) {
-			result.Skipped++
-			continue
+		return result, nil
```
[the remark this answers is not in the document]
Comment (mrcfps) [reply internal/reviewer/runner.go]: @chatgpt-codex-connector This has been fixed in ec83559.

Updated DiscoverPullRequest in internal/reviewer/runner.go to evaluate every reviewer loop matching the PR and stop only after one enqueues, and added a targeted regression test in internal/reviewer/runner_test.go.




<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>

**Comment by mrcfps** [comment]
**Looper fixer round complete** — ec83559

- ✅ Review comment on `internal/reviewer/runner.go` (@chatgpt-codex-connector) — [thread](https://github.com/nexu-io/looper/pull/375#discussion_r3252516515)
  - Updated internal/reviewer/runner.go so follow-up discovery marks a PR as seen only after discoverExistingReviewerLoop actually appends a queue item. That preserves eligible manual follow-up loops when an earlier automatic loop is skipped.
- ✅ Review comment on `internal/reviewer/runner.go` (@chatgpt-codex-connector) — [thread](https://github.com/nexu-io/looper/pull/375#discussion_r3252516516)
  - Updated DiscoverPullRequest in internal/reviewer/runner.go to evaluate every reviewer loop matching the PR and stop only after one enqueues, and added a targeted regression test in internal/reviewer/runner_test.go.


<sub>🔁 Powered by <a href="https://github.com/nexu-io/looper">Looper</a> · runner=fixer · agent=opencode · An autonomous AI dev team for your GitHub repos.</sub>
